src/covid_model_seiir_pipeline/lib/cli_tools/performance_logger/performance_logger.py

```diff
@@ -0,0 +1,47 @@
+from collections import defaultdict
+import time
+
+from loguru import logger
+
+
+class TaskPerformanceLogger:
```

Comment:
This logger is a little bit dumb.  It does automatic context switching, which means that if log messages go missing, we'll over allocate time to subtasks that aren't correct.  Not super worried about this in practice and doing explicit context switching would double the logging statements.